### PR TITLE
Added add_child_screen and remove_child_screen methods.

### LINKED ISSUE #463: Add a way to add a child view controller
I'd like to add an API (in PM 2.x) for adding a child screen to a screen. Here's an example when it would be useful:

https://github.com/jamonholmgren/promotion-motion-kit/blob/master/app/screens/home_screen.rb#L12-L15

Some thoughts I had on how it could look:

```ruby
@sub_screen = SubScreen.new
add @sub_screen
child @sub_screen
add_child @sub_screen
add_screen @sub_screen
add_child_screen @sub_screen
```

Thoughts?

### PR DESCRIPTION
**Pull request by jamonholmgren**
Closes #463. Soliciting feedback. Includes docs and tests.

### CODE REVIEW

File: lib/ProMotion/screen/screen_module.rb [hunk 1]
```diff
@@ -25,41 +25,6 @@ def modal?
       self.modal == true
     end
 
-    def resolve_title
```
Comment (jamonholmgren) [inline lib/ProMotion/screen/screen_module.rb]: Made these methods private, like they should have been. No other changes, just moved them to the private method section.

File: lib/ProMotion/screen/screen_module.rb [hunk 2]
```diff
@@ -172,10 +137,60 @@ def frame
...
+      screen.didMoveToParentViewController(self) # Required
+      screen
+    end
+
+    def remove_child_screen(screen)
+      screen.parent_screen = nil
+      screen.willMoveToParentViewController(nil) # Required
+      screen.removeFromParentViewController
+      screen
+    end
```
Comment (jamonholmgren) [inline lib/ProMotion/screen/screen_module.rb]: This section is the main addition.

File: lib/ProMotion/screen/screen_module.rb [hunk 3]
```diff
@@ -172,10 +137,60 @@ def frame
...
+      UIApplication.sharedApplication.setStatusBarHidden(hidden, withAnimation:self.class.status_bar_animation)
+    end
+
+    def status_bar_style(style)
+      UIApplication.sharedApplication.setStatusBarStyle(style)
+    end
+
     def apply_properties(args)
-      reserved_args = [ :nav_bar, :hide_nav_bar, :hide_tab_bar, :animated, :close_all, :in_tab, :in_detail, :in_master, :to_screen ]
+      reserved_args = [ :nav_bar, :hide_nav_bar, :hide_tab_bar, :animated, :close_all, :in_tab, :in_detail, :in_master, :to_screen, :toolbar ]
```
Comment (jamonholmgren) [inline lib/ProMotion/screen/screen_module.rb]: Added `:toolbar` because I was getting warnings in the tests.

**Comment by ryanlntn** [comment]
:+1: Looks good to me.

**Comment by markrickert** [comment]
Fixed `master` merge conflict introduced by #670.

welcome

**Pull request merged**